packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -81,9 +81,32 @@ class _KeyboardClientImpl implements mojom.KeyboardClient {
   }
 }
 
+/// Configurable state of an input field.
+class InputValue {
+  const InputValue({ this.text: '', this.selection });
+
+  /// The current text being edited.
+  final String text;
+
+  /// The range of text that is currently selected.
+  final TextSelection selection;
+
+  static const InputValue empty = const InputValue();
+
+  bool operator ==(Object other) {
+    if (other.runtimeType != runtimeType)
+      return false;
+    InputValue otherValue = other;
+    return (otherValue.text == text) &&
+           (otherValue.selection == selection);
```

Comment:
This won't work because TextSelection (via TextRange) doesn't have an operator ==.